dev-docs/bidder-adaptor.md

```diff
@@ -495,9 +495,102 @@ if (bid.mediaType === 'video' || (videoMediaType && context !== 'outstream')) {
+
+Use case 1
+```
+AdUnit config
+{
+  ...
+  adPodDuration: 300,
+  durationRangeSec: [15, 30]
+  ...
+}
+
+Algorithm
+# of placements = adPodDuration / MIN_VALUE(durationRangeSec)
+
+Each placement set max duration:
+placement.video.maxduration = MAX_VALUE(durationRangeSec)
+
+Example:
+# of placements : 300 / 15 = 20.
+placement.video.maxduration = 30 (all placements the same)
+
+Your endpoint responds with:
+10 bids with 30 seconds duration
+10 bids with 15 seconds duration
+```
+
+In Use case 1, you are asking endpoint to respond with 20 bids between min duration 0 and max duration 30 seconds. If you get bids with duration which does not match duration in `durationRangeSec` array, Prebid will modify the duration and use new value to send bids to Ad server.
+
+Prebid creates virtual duration buckets based on `durationRangeSec` value. Prebid will
+  - round the duration to the next highest specified duration value based on adunit. If the duration is above a range within a set buffer, that bid falls down into that bucket. (eg if `durationRangeSec` was [5, 15, 30] -> 2s is rounded to 5s; 17s is rounded back to 15s; 18s is rounded up to 30s)
+  - reject bid if the bid is above the range of the listed durations (and outside the buffer)
+  
+Prebid will set the rounded duration value in the `bid.video.durationBucket` field for accepted bids
+
+Use case 2
+```
+AdUnit config
+{
+  ...
+  adPodDuration: 300,
+  durationRangeSec: [15, 30],
+  requireExactDuration: true
+  ...
+}
+
+Algorithm
+# of placements = MAX_VALUE(adPodDuration/MIN_VALUE(allowedDurationsSec), durationRangeSec.length) 
+
+Each placement:
+placement.video.minduration = durationRangeSec[i]
+placement.video.maxduration = durationRangeSec[i]
+
+Example:
+# of placements : MAX_VALUE( (300 / 15 = 20), 2) == 20
+
+20 / 2 = 10 placements: 
+placement.video.minduration = 15
+placement.video.maxduration = 15
+
+20 / 2 = 10 placements: 
+placement.video.minduration = 30
+placement.video.maxduration = 30
+
+Your endpoint responds with:
+10 bids with 30 seconds duration
+10 bids with 15 seconds duration
+```
+
+In Use case 2 `requireExactDuration` is set to true and hence Prebid will only select bids that exactly match duration in `durationRangeSec` (don't round at all).
+
+In both use cases, adapter is requesting bid responses for 20 placements in one single http request. You can split these into chunks depending on your endpoint's capacity.
+
+Adapter must add following new properties to bid response
+
+{% highlight js %}
+{
+  meta: {
+    iabSubCatId: '<iab sub category>', // only needed if you want to ensure competitive separation
+  },
+  video: {
+    context: 'adpod',
+    durationSeconds: 30
+  }
+}
+{% endhighlight %}
+
+
+Appnexus Adapter uses above explained approach. You can refer [here](https://github.com/prebid/Prebid.js/blob/master/modules/appnexusBidAdapter.js)
```

Comment:
Is there a function or sub-section in the adapter code that we can point them to so it's easier for the reader to reference the point we want to make?  The current link just goes to the top of the adapter, so it's extra effort for the reader to discern the code in question.

Reply:
Everything we explained here is not sitting in one place in adapter. It is scattered at various functions. It will be the same for other adapters as well. 

Hence just sharing the Appnexus adapter file path. Very little extra effort for developer implementing this.